File: primes/ruby/compute.rb
PRIME_COUNT_STR = ENV["PRIME_COUNT"]
if PRIME_COUNT_STR == nil then
  puts "Please set PRIME_COUNT"
  exit(-1)
end
PRIME_COUNT = PRIME_COUNT_STR.to_i
BENCH_DEBUG = ENV["BENCH_DEBUG"]

puts "begin"
$stdout.flush

primes = []
number = 2
while primes.size < PRIME_COUNT do
  is_multiple = primes.detect { |p| number % p == 0 }
  if not is_multiple then
    primes << number
  end
  number += 1
end

if BENCH_DEBUG == "true" then
  primes.each do |prime|
    puts prime
  end
end

puts "end"
$stdout.flush

File: primes/lua-jit/compute.lua
local PRIME_COUNT = assert(
    os.getenv("PRIME_COUNT"),
    "Please set PRIME_COUNT environment variable."
)
PRIME_COUNT = tonumber(PRIME_COUNT)

local primes = {}
local function is_prime(number)
    for _, prime in ipairs(primes) do
        if number % prime == 0 then
            return false
        end
    end
    return true
end

print("begin")
io.stdout:flush()
local number = 2
while #primes < PRIME_COUNT do
    if is_prime(number) then
        print(number)
        table.insert(primes, number)
    end
    number = number + 1
end

print("end")
io.stdout:flush()
